extensions/ql-vscode/src/view/model-editor/ModelTypeTextbox.tsx

```diff
@@ -0,0 +1,71 @@
+import type { ChangeEvent } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import type { ModeledMethod } from "../../model-editor/modeled-method";
+import { VSCodeTextField } from "@vscode/webview-ui-toolkit/react";
+import { useDebounceCallback } from "../common/useDebounceCallback";
+
+type Props = {
+  modeledMethod: ModeledMethod | undefined;
```

Comment:
Since we're only using this component when the `modeledMethod.type === "type"`, I think this can be simplified to:

```suggestion
  modeledMethod: TypeModeledMethod;
```

This means we also remove some checks related to checking whether the `modeledMethod.type === "type"` in the component itself.

Reply:
Good point, thanks!